### DISCUSSION TITLE
m4a: Longer Analysis Time?

### DISCUSSION THREAD

**Discussion by Vollpflock** [Mattk70/Chirpity-Electron#89] [General]
I'm still around, just busy setting up recording stations and getting them up and running. I've made the switch from 22kHz WAV 16 bit to m4a 44kHz. It's great as it only occupies 1/5 of the storage space while maintaining comparable quality. However, I've noticed that it takes three times longer to analyze m4a files. For instance, it takes about 90 seconds to analyze a one-hour m4a file compared to just around 31 seconds for a one-hour WAV file on my M1 Mac using 9 threads and a batch size of 64. Is this expected?

**Comment by Mattk70** [comment]
Hello again, welcome back ;) 

Yes, it will take longer to analyse m4a files. They need to be decompressed on the fly before the audio is sent to the model. Decompression is done by the single thread ffmpeg process. It's a bottleneck when I use the webgl backend on my PC, but doesn't make a great deal of difference with the CPU backend on my Mac. On my M2, a 3h 22m WAV file (16bit, 44.1KHz mono)  is processed in 80 seconds. The same file, exported to m4a in Audacity with 160 bitrate - takes 90 seconds. I'm using the default CPU settings - all threads (8) and batch size 32.

**Comment by Vollpflock** [comment]
Thank you. That's strange. Maybe I need to test this on an M2 machine, but it shouldn't make such a big difference.

**Comment by Vollpflock** [comment]
So I did some testing on my M1 with m4a files, each 1 hour long. M4a files seem to require different settings. 

All CPU. 
7 threads 128 batch size: 180+ seconds (had to quit)
7 threads 64 batch size: ≈90 seconds
7 threads 32 batch size: ≈55 seconds
7 threads 16 batch size: ≈40 seconds

10 threads 16 batch size: ≈40 seconds

**Reply by Mattk70** [reply]
I'm glad you found a solution, Just speculating, but it may be that there are increased memory requirements for m4a processing, and the larger batch sizes are hitting a limit - either on your Mac, or more likely with the chromium sandbox that underlies the Chirpity application.

**Comment by Vollpflock** [comment]
That's an interesting idea. I just tried analyzing around 50 1hour long m4a files at 7 threads 16 batch size and it crashed at at the end. I have over 10% storage free on my mac (72 gb / 500 gb), which isn't that much, but should do the job. Maybe that's the issue..
